### PR TITLE
[OSASINFRA] Add an option to set the project name

### PR DESCRIPTION
**Pull request by itzikb-redhat**
We use a project name for the shiftstackclient as well as the OCP on top of OSP cluster name

Also:
  - Delete the shiftstackclient pod before creating it to make sure we are not reusing an old one
  - Modify the installation artifacts as it was changed in shiftstack-qa

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: roles/shiftstack/defaults/main.yml [hunk 1]
```diff
@@ -20,12 +20,14 @@
 cifmw_shiftstack_artifacts_dir: "{{ cifmw_shiftstack_basedir }}/artifacts"
 cifmw_shiftstack_basedir: "{{ cifmw_basedir | default(ansible_user_dir ~ '/ci-framework-data') }}/tests/shiftstack"
 cifmw_shiftstack_client_pod_image: "quay.io/shiftstack-qe/shiftstack-client:latest"
-cifmw_shiftstack_client_pod_name: "shiftstackclient"
+cifmw_shiftstack_client_pod_name: "shiftstackclient-{{ cifmw_shiftstack_project_name }}"
 cifmw_shiftstack_client_pod_namespace: "openstack"
+cifmw_shiftstack_cluster_name: "shiftstack"
```
Comment (rlobillo) [inline roles/shiftstack/defaults/main.yml:25]: I think it's the other way around. The cluster name is 'ostest' and the project name is 'shiftstack'

**Comment by imatza-rh** [comment]
> artifacts/
├── must-gather-verification
├── installation
│   └── shiftstack                       # `user_cloud_installation_dir`
│       ├── install-config.yaml
│       ├── openstack                    # `osp_config_dir`
│       │   ├── clouds.yaml
│       │   └── shiftstackrc
│       └── ostest                       # `ocp_installation_dir`


^ 1196079: Set the installation dir inside the artifacts dir | https://review.gerrithub.io/c/shiftstack/shiftstack-qa/+/1196079

File: roles/shiftstack/tasks/pre_test_shiftstack.yml [hunk 2]
```diff
@@ -13,6 +13,13 @@
 # WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 # License for the specific language governing permissions and limitations
 # under the License.
+- name: Remove the shiftstackclient pod if exists
+  kubernetes.core.k8s:
+    state: absent
+    api_version: v1
+    kind: Pod
+    namespace: "{{ cifmw_shiftstack_client_pod_namespace }}"
+    name: "{{ cifmw_shiftstack_client_pod_name + '_pod.yml' }}"
```
Comment (imatza-rh) [inline roles/shiftstack/tasks/pre_test_shiftstack.yml:22]:
```suggestion
    name: "{{ cifmw_shiftstack_client_pod_name }}"
```
I think we should use the pod name here.

@eurijon wdyt about adding a var for the `cifmw_shiftstack_client_pod_name_pod.yml` file?

**Review by imatza-rh — COMMENTED** [review]
Can you please add a reference to the commit-msg
```
Modify the installation artifacts as it was changed in shiftstack-qa
```
```
artifacts/
├── must-gather-verification
├── installation
│   └── shiftstack                       # `user_cloud_installation_dir`
│       ├── install-config.yaml
│       ├── openstack                    # `osp_config_dir`
│       │   ├── clouds.yaml
│       │   └── shiftstackrc
│       └── ostest                       # `ocp_installation_dir`

^ 1196079: Set the installation dir inside the artifacts dir | https://review.gerrithub.io/c/shiftstack/shiftstack-qa/+/1196079
```

**Comment by itzikb-redhat** [comment]
```
[cloud-admin@shiftstackclient-shiftstack101 ~]$ oc get nodes
NAME                             STATUS   ROLES                  AGE    VERSION
ostest101-nkprb-master-0         Ready    control-plane,master   147m   v1.29.5+29c95f3
ostest101-nkprb-master-1         Ready    control-plane,master   147m   v1.29.5+29c95f3
ostest101-nkprb-master-2         Ready    control-plane,master   148m   v1.29.5+29c95f3
ostest101-nkprb-worker-0-h2m2x   Ready    worker                 101m   v1.29.5+29c95f3
ostest101-nkprb-worker-0-lbrv6   Ready    worker                 108m   v1.29.5+29c95f3
[cloud-admin@shiftstackclient-shiftstack101 ~]$ oc get machines -A
NAMESPACE               NAME                             PHASE     TYPE     REGION      ZONE   AGE
openshift-machine-api   ostest101-nkprb-master-0         Running   master   regionOne   nova   158m
openshift-machine-api   ostest101-nkprb-master-1         Running   master   regionOne   nova   158m
openshift-machine-api   ostest101-nkprb-master-2         Running   master   regionOne   nova   158m
openshift-machine-api   ostest101-nkprb-worker-0-h2m2x   Running   worker   regionOne   nova   131m
openshift-machine-api   ostest101-nkprb-worker-0-lbrv6   Running   worker   regionOne   nova   131m
[cloud-admin@shiftstackclient-shiftstack101 ~]$ openstack server list --os-cloud shiftstack101
+---------------------------------+--------------------------------+--------+----------------------------------+-----------------------+--------+
| ID                              | Name                           | Status | Networks                         | Image                 | Flavor |
+---------------------------------+--------------------------------+--------+----------------------------------+-----------------------+--------+
| 5b0e44b0-6002-4ac3-855c-        | ostest101-nkprb-worker-0-h2m2x | ACTIVE | k8s-clusterapi-cluster-          | ostest101-nkprb-rhcos | worker |
| 69bd214797f6                    |                                |        | openshift-cluster-api-guests-    |                       |        |
|                                 |                                |        | ostest101-nkprb=10.196.2.98      |                       |        |
| 60562a55-8d3d-441e-b1d8-        | ostest101-nkprb-worker-0-lbrv6 | ACTIVE | k8s-clusterapi-cluster-          | ostest101-nkprb-rhcos | worker |
| b28c16682643                    |                                |        | openshift-cluster-api-guests-    |                       |        |
|                                 |                                |        | ostest101-nkprb=10.196.1.229     |                       |        |
| 7f85e30a-d5be-46c2-a217-        | ostest101-nkprb-master-2       | ACTIVE | k8s-clusterapi-cluster-          | ostest101-nkprb-rhcos | master |
| fe11b6025369                    |                                |        | openshift-cluster-api-guests-    |                       |        |
|                                 |                                |        | ostest101-nkprb=10.196.3.137     |                       |        |
| 43610fc0-b528-4427-9edc-        | ostest101-nkprb-master-0       | ACTIVE | k8s-clusterapi-cluster-          | ostest101-nkprb-rhcos | master |
| e806add0709f                    |                                |        | openshift-cluster-api-guests-    |                       |        |
|                                 |                                |        | ostest101-nkprb=10.196.1.188     |                       |        |
| 8c815067-d054-40eb-9f73-        | ostest101-nkprb-master-1       | ACTIVE | k8s-clusterapi-cluster-          | ostest101-nkprb-rhcos | master |
| 1a00ee0b3e33                    |                                |        | openshift-cluster-api-guests-    |                       |        |
|                                 |                                |        | ostest101-nkprb=10.196.0.101     |                       |        |
+---------------------------------+--------------------------------+--------+----------------------------------+-----------------------+--------+
```

File: roles/shiftstack/tasks/test_shiftstack.yml [hunk 3]
```diff
@@ -82,19 +82,12 @@
 
   always:
     - name: Retrieve artifacts and installation files from the pod
```
Comment (eurijon) [inline roles/shiftstack/tasks/test_shiftstack.yml:84]: As the there is no loop any more I would remove the "installation files" from the task name.

File: roles/shiftstack/tasks/test_shiftstack.yml [hunk 4]
```diff
@@ -82,19 +82,12 @@
...
-      vars:
-        directory_list:
-          - controller_dir: "{{ cifmw_shiftstack_artifacts_dir }}"
-            pod_dir: "{{ cifmw_shiftstack_shiftstackclient_artifacts_dir }}"
-          - controller_dir: "{{ cifmw_shiftstack_installation_dir }}"
-            pod_dir: "{{ cifmw_shiftstack_shiftstackclient_installation_dir }}"
       block:
         - name: Create the directories for the artifacts and installation files
```
Comment (eurijon) [inline roles/shiftstack/tasks/test_shiftstack.yml:86]: Same here

File: roles/shiftstack/tasks/test_shiftstack.yml [hunk 5]
```diff
@@ -82,19 +82,12 @@
...
           ansible.builtin.file:
-            path: "{{ item.controller_dir }}"
+            path: "{{ cifmw_shiftstack_artifacts_dir }}"
             state: directory
             mode: "0755"
-          loop: "{{ directory_list }}"
 
         - name: Copy the artifacts and installation files from the pod '{{ cifmw_shiftstack_client_pod_name }}'
```
Comment (eurijon) [inline roles/shiftstack/tasks/test_shiftstack.yml:92]: Same here

File: roles/shiftstack/tasks/pre_test_shiftstack.yml (hunk 2 above)
Comment (eurijon) [reply roles/shiftstack/tasks/pre_test_shiftstack.yml]: We can add a param for that indeed (same would apply for the PVC yaml file). The appended `_pod.yml` and `_pvc.yaml` strings are just for the yml files names, so we can easily identify each file when looking at the artifacts. Having them in params can make the code cleaner.

**Review by eurijon — COMMENTED** [review]
Overall looks good! added small comments

File: roles/shiftstack/README.md [hunk 6]
```diff
@@ -7,7 +7,9 @@ Role for triggering Openshift on Openstack QA automation (installation and tests
 * `cifmw_shiftstack_client_pod_name`: (*string*) Pod name for the pod running the Openshift installer and tests. Defaults to `shiftstackclient`.
 * `cifmw_shiftstack_client_pod_namespace`: (*string*) The namespace where the `cifmw_shiftstack_client_pod_name` will be deployed. Defaults to `openstack`.
 * `cifmw_shiftstack_client_pod_image`: (*string*) The image for the container running in the `cifmw_shiftstack_client_pod_name` pod. Defaults to `quay.io/shiftstack-qe/shiftstack-client:latest`.
-* `cifmw_shiftstack_installation_dir`: (*string*) Directory to place installation files. Defaults to `"{{ cifmw_shiftstack_basedir }}/installation"`.
+* `cifmw_shiftstack_cluster_name`: (*string*) The underlying Openshift cluster name. Defaults to `ostest`.
```
[the remark this answers is not in the document]
Comment (itzikb-redhat) [reply roles/shiftstack/README.md:10]: Done

File: roles/shiftstack/README.md [hunk 7]
```diff
@@ -7,7 +7,9 @@ Role for triggering Openshift on Openstack QA automation (installation and tests
...
+* `cifmw_shiftstack_installation_dir`: (*string*) Directory to place installation files. Defaults to `"{{ cifmw_shiftstack_shiftstackclient_artifacts_dir }}/installation"`.
+* `cifmw_shiftstack_project_name`: (*string*) The Openstack project name name. Defaults to `shiftstack`.
```
[the remark this answers is not in the document]
Comment (itzikb-redhat) [reply roles/shiftstack/README.md:12]: Done

File: roles/shiftstack/tasks/test_shiftstack.yml (hunk 3 above)
Comment (itzikb-redhat) [reply roles/shiftstack/tasks/test_shiftstack.yml:84]: Done

File: roles/shiftstack/tasks/test_shiftstack.yml [hunk 8]
```diff
@@ -81,28 +81,20 @@
...
-      vars:
-        directory_list:
-          - controller_dir: "{{ cifmw_shiftstack_artifacts_dir }}"
-            pod_dir: "{{ cifmw_shiftstack_shiftstackclient_artifacts_dir }}"
-          - controller_dir: "{{ cifmw_shiftstack_installation_dir }}"
-            pod_dir: "{{ cifmw_shiftstack_shiftstackclient_installation_dir }}"
+    - name: Retrieve artifacts from the pod
       block:
-        - name: Create the directories for the artifacts and installation files
+        - name: Create the directories for the artifacts
```
Comment (eurijon) [inline roles/shiftstack/tasks/test_shiftstack.yml:86]:
```suggestion
        - name: Create the directory for the artifacts
```

Comment (itzikb-redhat) [reply roles/shiftstack/tasks/test_shiftstack.yml:86]: Done

**Comment by cjeanner** [comment]
/approve

molecule for the role is green - can't really say more, this role is dedicated for a specific usecase afaik. It won't break "us" (RHOSO).

**Pull request merged**